observability/custom-metrics-autoscaling/prometheus-to-sd-gmp/custom-metrics-prometheus-sd-gmp.yaml

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START gke_prometheus_to_sd_gmp_custom_metrics_prometheus_sd_gmp_deployment_custom_metric_prometheus_sd_gmp]
+# [START container_custom_metrics_prometheus_deployment]
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    run: custom-metric-prometheus-sd-gmp
+  name: custom-metric-prometheus-sd-gmp
+  namespace: default
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      run: custom-metric-prometheus-sd-gmp
+  template:
+    metadata:
+      labels:
+        run: custom-metric-prometheus-sd-gmp
+    spec:
+      containers:
+      # sample container generating custom metrics
+      - name: prometheus-dummy-exporter
+        image: us-docker.pkg.dev/google-samples/containers/gke/prometheus-dummy-exporter:v0.2.0
+        command: ["./prometheus-dummy-exporter"]
+        args:
+        - --metric-name=custom_prometheus
+        - --metric-value=40
+        - --port=8080
+# [END container_custom_metrics_prometheus_deployment]
+# [END gke_prometheus_to_sd_gmp_custom_metrics_prometheus_sd_gmp_deployment_custom_metric_prometheus_sd_gmp]
+---
+# [START gke_prometheus_to_sd_gmp_custom_metrics_prometheus_sd_gmp_pod_monitoring_custom_metric_prometheus_sd_gmp]
+apiVersion: monitoring.googleapis.com/v1
+kind: PodMonitoring
+metadata:
+  name: "prometheus-to-prometheus"
+spec:
+  selector:
+    matchLabels:
+      run: "custom-metric-prometheus-sd-gmp"
+  endpoints:
+  - port: 8080
+    path: /metrics
+    interval: 60s
```

Comment:
I know that this demo application exports a constant value, but what about setting this interval to a shorter period (e.g., 15s)? So people can have a faster autoscaling reaction time if they decide to customize the underlying application and reuse this PodMonitoring object.

Reply:
Done.